msg.go

```diff
@@ -707,19 +708,19 @@ func (dns *Msg) PackBuffer(buf []byte) (msg []byte, err error) {
 	if dns.Rcode < 0 || dns.Rcode > 0xFFF {
 		return nil, ErrRcode
 	}
+	rcode = dns.Rcode & 0xF
 	if dns.Rcode > 0xF {
 		// Regular RCODE field is 4 bits
 		opt := dns.IsEdns0()
 		if opt == nil {
 			return nil, ErrExtendedRcode
 		}
 		opt.SetExtendedRcode(uint8(dns.Rcode >> 4))
-		dns.Rcode &= 0xF
 	}
 
 	// Convert convenient Msg into wire-like Header.
 	dh.Id = dns.Id
-	dh.Bits = uint16(dns.Opcode)<<11 | uint16(dns.Rcode)
+	dh.Bits = uint16(dns.Opcode)<<11 | uint16(rcode)
```

Comment:
I think it might be cleaner to just put the `dns.Rcode & 0xF` here in the `uint64(...)`.

Reply:
I don't have strong feelings. If this is is needed to get this merged (assuming the patch logic is accepted), I will update the PR.

Reply:
might be a tad bit cleaner, can you also add a comment above it on why this needed?